org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchTests.java

```diff
@@ -4593,4 +4576,108 @@ public void testAnonymousTypeMethodReferenceJarSearchGh375() throws Exception {
 		deleteProject("P");
 	}
 }
+/*
+ * Test that we can find method references called in an anonymous type defined in a method.
+ * https://github.com/eclipse-jdt/eclipse.jdt.core/issues/432
+ */
+public void testAnonymousTypeMethodReferenceSearchGh432() throws Exception {
+	String snippet1 = "package p;\n" +
+			"public class TestGh432 {\n" +
+			"  public static void main(String[] args) {\n" +
+			"    class R implements Runnable {\n" +
+			"      public void run() {\n" +
+			"        java.util.stream.Stream.empty().filter(TestGh432::missingReference).count();\n" +
+			"      }\n" +
+			"    }\n" +
+			"    new R().run();\n" +
+			"  }\n" +
+			"  public static boolean missingReference() {\n" +
```

Comment:
Is it intentional that the method `missingReference` does not match the signature of the Predicate that needs to be passed to Stream.filter?

Reply:
> Is it intentional that the method `missingReference` does not match the signature of the Predicate that needs to be passed to Stream.filter?

No, this is an oversight. I'll adjust it.